smoke-tests/spec/external_dns_helper.rb

```diff
@@ -1,31 +1,32 @@
-def create_ingress
+def create_ingress(namespace, ingress_name, fixture_name)
+  sleep 1
   apply_template_file(
     namespace: namespace,
-    file: "spec/fixtures/external-dns-ingress.yaml.erb",
+    file: fixture_name,
     binding: binding
   )
-  wait_for(namespace, "ingress", "ingress-external-dns", 60)
+  wait_for(namespace, "ingress", ingress_name, 60)
 end
 
-def delete_ingress(namespace)
-  `kubectl delete ingress ingress-external-dns -n #{namespace}`
-  sleep 10
+def delete_ingress(namespace, ingress_name)
+  `kubectl delete ingress #{ingress_name} -n #{namespace}`
 end
 
-def get_ingress_enpoint
-  `kubectl get ingress integration-test -o json -o jsonpath='{.status.loadBalancer.ingress[0].hostname}'`
+def get_ingress_enpoint(namespace, ingress_name)
+  `kubectl get ingress #{ingress_name} -n #{namespace} -o json -o jsonpath='{.status.loadBalancer.ingress[0].hostname}'`
 end
 
-def delete_A_record(zone_id, zone_name, domain_name)
+def delete_A_record(zone_id, zone_name, domain_name, namespace, ingress_name)
+  sleep 1
   client = Aws::Route53::Client.new
 
   a_record = {
     action: "DELETE",
     resource_record_set: {
-      name: zone_name,
+      name: domain_name,
       alias_target: {
         "hosted_zone_id": "ZD4D7Y8KGAS4G",
```

Comment:
Where does `ZD4D7Y8KGAS4G` come from?

Reply:
It seems to be the zone where all the ELB domains are defined in eu-west-2, by AWS.
https://docs.aws.amazon.com/general/latest/gr/rande.html